### ISSUE TITLE
0.7.1 my test started freezing

### ISSUE THREAD

**Issue by jtakahashi0604** [amatsuda/database_rewinder#46]
Hello, I like and use this gem.
Just updated to version from `0.6.5` to `0.7.1` and my test started freezing.

![image](https://cloud.githubusercontent.com/assets/7970164/20638091/35118b4e-b3df-11e6-9f0b-b2c57b35e5de.png)

```sql
select t_b.trx_mysql_thread_id blocking_id,
       t_w.trx_mysql_thread_id requesting_id,
       p_b.HOST blocking_host,
       p_w.HOST requesting_host,
       l.lock_table lock_table,
       l.lock_index lock_index,
       l.lock_mode lock_mode,
       p_w.TIME seconds,
       p_b.INFO blocking_info,
       p_w.INFO requesting_info
from information_schema.INNODB_LOCK_WAITS w,
     information_schema.INNODB_LOCKS l,
     information_schema.INNODB_TRX t_b,
     information_schema.INNODB_TRX t_w,
     information_schema.PROCESSLIST p_b,
     information_schema.PROCESSLIST p_w
where w.blocking_lock_id = l.lock_id
  and w.blocking_trx_id = t_b.trx_id
  and w.requesting_trx_id = t_w.trx_id
  and t_b.trx_mysql_thread_id = p_b.ID
  and t_w.trx_mysql_thread_id = p_w.ID
order by requesting_id,
         blocking_id
\G
```

```
*************************** 1. row ***************************
    blocking_id: 1652
  requesting_id: 1654
  blocking_host: localhost
requesting_host: localhost
     lock_table: `foo_test`.`bars`
     lock_index: PRIMARY
      lock_mode: X
        seconds: 4
  blocking_info: NULL
requesting_info: DELETE FROM bars
1 row in set (0.01 sec)
```

freezing part code.
https://github.com/amatsuda/database_rewinder/blob/2e0dcab644a36436b91e70773ed714af666e3d6b/lib/database_rewinder/multiple_statements_executor.rb#L32

## environment
I use rails `5.0.0.1` and ruby `2.3.1` and MySQL `5.6.25`.

## solution(temporary)
```
config.use_transactional_fixtures = false
```

references:
http://mrdanadams.com/2012/rails-rspec-capybara-mysql-deadlocks

I suppose database deadlocks ploblem.

Thank you for always good job.

**Comment by amatsuda** [comment]
Yep, that is a known behavior. As you found out, database_rewinder causes deadlock with transactional tests under some conditions (here's a brief description written in Japanese https://www.facebook.com/akinori.musha/posts/10154719325004839?comment_id=10154720640984839 ).

Firstly, TBH I still don't understand why would you need to use database_rewinder where transactional tests are working.
Disabling transactional tests is not a temporary workaround. It's a required configuration to properly use database_{cleaner,rewinder}.

Anyway it's true that we need some improvements on our library.
So here's actually a possible solution that I wrote 2 weeks ago: 3c169f91b301aad89edbf117a5aeec97bf2156bd
This patch adds `multiple: false` option to `clean()` and `clean_all()` that allows users to run non-multiple deletion queries if they're really sure what they are doing and really need to do it.

I haven't committed it to master because I'm not sure if it's a good approach. I'd love to hear your thoughts.

**Comment by jtakahashi0604** [comment]
Thanks for your prompt reply.

I read brief description and understand.

database_rewinder setting in my project prior to immediately need,
But looking ahead use it.

Thanks for good patch: 3c169f9

Is not needed `multiple: false` option.
I am happy if written document `#### 1. Properly understand what use_transactional_tests means, and consider turning it off`.

**Comment by amatsuda** [comment]
This should be fixed in 0.8.0 release. I'm closing this issue for now, but feel free to open another PR or issue if you're still seeing this problem.

**Comment by jtakahashi0604** [comment]
It is not problem, Thanks you for support.
